05/app.js

```diff
@@ -1 +1,16 @@
-const numbers = [1, 2, 3, 4, 5, 6, 7];
\ No newline at end of file
+const numbers = [1, 2, 3, 4, 5, 6, 7];
+
+const evenNumbersSum = numbers.filter(function(number) {
+    return number % 2 === 0;
+})
+
+.reduce(function(sum, number) {
+    return sum + number;
+});
+
+console.log(evenNumbersSum);
+
+
+// jesli dobrze to rozumiem to .reduce() dodaje do siebie wszystkie elementy tablicy i zwraca je jako jedną wartość??
+
+// czy raczej dodaje wszystkie elementy tablicy do pierwszego elementu (accumulator)??
```

Comment:
Oba są prawdziwe :P

`reduce` działa tak, że to co w `return` wpada do pierwszego parametru i operacja jest wykonywana aż do przejścia po każdym elemencie tablicy, który jest w drugim parametrze. Warto zapisać trochę inaczej reduce tj. dodać 0 jako pierwszy element w `sum` - wtedy pewnie będzie to czytelniejsze jak wyświetlimy zawartość parametrów tj.
```
.reduce(function(sum, number) {
    console.log(sum, number);
    return sum + number;
}, 0);
```